### PR TITLE
Add url encoded middleware and csrf middleware

### PR DESCRIPTION
**Pull request by rnpenguin**

### CODE REVIEW

File: libraries/hatch-server-middleware/package.json [hunk 1]
```diff
@@ -24,7 +24,12 @@
     "express": "^4",
     "utf-8-validate": "^5",
     "winston": "^3",
-    "ws": "^7"
+    "ws": "^7",
+    "@types/cookie": "0.4.1",
```
Comment (kmcgowan) [inline libraries/hatch-server-middleware/package.json:28]: peer dependencies should also be dev dependencies

File: libraries/hatch-server-middleware/src/CsrfMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  public async register(app: Application) {
+    app.use(isCsrfSafe);
+  }
+}
+
+const isMethodSideEffectSafe = (method: string): boolean => {
+  return ['GET', 'HEAD', 'OPTIONS'].includes(method.toUpperCase());
+};
+
+const isCsrfSafe = (req: Request, res: Response, next: NextFunction) => {
```
Comment (kmcgowan) [inline libraries/hatch-server-middleware/src/CsrfMiddleware.ts:19]: Is this duplicated in the auth middleware? Can that call this?

File: libraries/hatch-server-middleware/package.json (hunk 1 above)
Comment (rnpenguin) [reply libraries/hatch-server-middleware/package.json]: updated to dependencies

File: libraries/hatch-server-middleware/src/CsrfMiddleware.ts (hunk 2 above)
Comment (rnpenguin) [reply libraries/hatch-server-middleware/src/CsrfMiddleware.ts]: removed duplication

**Review by kiltyj — APPROVED** [review]

**Pull request merged**